### PR TITLE
added more agency metrics

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Added the following agency-related metrics:
- `arangodb_agency_client_lookup_table_size`: current number of entries in agency client id lookup table. This gauge is available only on agent instances.
- `arangodb_agency_cache_callback_count`: current number of entries in agency cache callbacks table. This gauge will be effective on coordinators and DB servers.
- `arangodb_agency_callback_count`: current number of agency callbacks registered. This gauge will be effectiive on coordinators and DB servers.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] Backports required for: 3.7

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13221/

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 
+* Added the following agency-related metrics:
+  - `arangodb_agency_client_lookup_table_size`: current number of entries
+    in agency client id lookup table. This gauge is available only on
+    agent instances.
+  - `arangodb_agency_cache_callback_count`: current number of entries in
+    agency cache callbacks table. This gauge will be effective on coordinators
+    and DB servers.
+  - `arangodb_agency_callback_count`: current number of agency callbacks
+    registered. This gauge will be effecitive on coordinators and DB servers.
```
Comment (jsteemann) [inline CHANGELOG:12]:
```suggestion
    registered. This gauge will be effective on coordinators and DB servers.
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13270/

**Review by kvahed — APPROVED** [review]
LGTM

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13281/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**